### PR TITLE
create a real release instead of a draft

### PR DESCRIPTION
**Pull request by spencerdrak**
PR checklist:

- [x] Documentation is up-to-date
- [x] `changelog.d/<issue>.<type>` is a file with the _what_, _why_, and _how_ of the change.
  - \<issue> is `pa-312` (Linear ticket), `gh-1234` (GitHub issue), or `new-gizmo` (unique semantic name)
  - \<type> is `added`, `changed`, `fixed`, or `infra`.
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see the [Contribution guidelines](https://semgrep.dev/docs/contributing/how-to-contribute/)!

### CODE REVIEW

**Review by underyx — APPROVED** [review]

File: .dockerignore [hunk 1]
```diff
@@ -183,3 +183,6 @@ semgrep-core/bin
 
 # Ignore .history and be doomed to repeat it
```
Comment (underyx) [inline .dockerignore:184]: hehe

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -39,24 +28,20 @@ jobs:
...
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          upload_url: ${{ steps.create_release.outputs.upload_url }}
-          asset_path: ./semgrep-osx/artifacts.zip
-          asset_name: semgrep-${{ steps.get_version.outputs.VERSION }}-osx.zip
-          asset_content_type: application/zip
+        run: |
+          cp ./semgrep-osx/artifacts.zip semgrep-${{ steps.get_version.outputs.VERSION }}-osx.zip
+          gh release upload ${{ steps.version.outputs.tag }} semgrep-${{ steps.get_version.outputs.VERSION }}-osx.zip
```
Comment (brendongo) [inline .github/workflows/release.yml:35]: Can you release upload before creating a draft release?

Comment (spencerdrak) [reply .github/workflows/release.yml:35]: Just ran a quick test - no, the release must exist (either draft or published) before you can upload assets to it.

Comment (spencerdrak) [reply .github/workflows/release.yml:35]: However, `start-release.yml` creates the draft with the changlog etc, so this succeeds. There's a TON of oddities in how GH handles this (both from the actions side and the releases side) and this was the cleanest way I could find to do this.

**Pull request merged**